examples/rotating-voronoi/src/index.ts

```diff
@@ -0,0 +1,222 @@
+import { updateDOM } from "@thi.ng/transducers-hdom";
+import {
+    AppState,
+    scaleStream,
+    animationStream,
+    frameStreamConditional,
+    frameStream,
+    keyStream,
+    keyStreamConditional,
+    mainStream
+} from "./stream-state";
+import {
+    map,
+    iterator1,
+    normRange,
+    comp,
+    mapcat,
+    iterator
+} from "@thi.ng/transducers";
+import { slider, checkbox } from "./controllers";
+import { download } from "./download";
+
+const edge = window.innerWidth * 0.7;
+const width = edge;
+const height = edge;
+const radius = (width / 2) * 0.8;
+const center = [width / 2, height / 2];
+
+const rndInt = (min: number, max: number) =>
+    Math.trunc(min + Math.random() * (max - min));
+
+const startingCircles: Array<[number, number, boolean]> = [
+    [radius / 1, rndInt(4, 20), true],
+    [radius / 2, rndInt(4, 20), false],
+    [radius / 4, rndInt(4, 20), true],
+    [radius / 8, rndInt(4, 20), false]
+];
+
+const pointsInCircle = (
+    _center: Vec,
+    _radius: number,
+    _num: number,
+    _angle: number
+) => [
+    ...iterator1(
+        map((index) =>
+            cartesian2([], [_radius, index * TAU + _angle], _center)
+        ),
+        normRange(_num, false)
+    )
+];
+
+scaleStream.next(1);
+frameStreamConditional.next(0);
+animationStream.next(false);
+const startKeyEvent: KeyboardEvent = document.createEvent("KeyboardEvent");
+startKeyEvent.initEvent("keyup");
+keyStreamConditional.next(startKeyEvent);
+
+const app = (...args: any) => (state: AppState) => appRender(state);
+mainStream.transform(
+    map(app(scaleStream, animationStream, frameStream, keyStream)),
+    updateDOM()
+);
+
+function computeVoronoi(state: AppState) {
+    const delta = state.frameValue / 100;
+    const doSave = state.keyValue === "s";
+
+    const opts: CubicOpts = {
+        breakPoints: false,
+        uniform: false,
+        scale: state.scaleValue
+    };
+
+    const startPoints = [
+        ...iterator(
+            comp(
+                map(([rad, density, clockwise]) =>
+                    pointsInCircle(
+                        center,
+                        rad,
+                        density,
+                        clockwise ? delta : PI - delta
+                    )
+                ),
+                mapcat((x) => x)
+            ),
+            startingCircles
+        )
+    ];
+
+    const mesh = new DVMesh();
+    mesh.addKeys(startPoints, 0.01);
+    const bounds = [[0, 0], [width, 0], [width, height], [0, height]];
+    const cells = mesh.voronoi(bounds);
+
+    const voronoi = [
+        rect([width, height], { fill: "black" }),
+
+        ...cells.map((cell) => {
+            const cell0 = new Polygon(simplify(cell, 0.01, true));
+
+            const cubics = asCubic(cell0, opts);
+            const paths = pathFromCubics(cubics);
+
+            return group({}, [
+                withAttribs(paths, { fill: "white", "stroke-width": 1 })
+            ]);
```

Comment:
`pathFromCubics` always only produces a single path, so technically, there's no need to create a group here. However, you could wrap this whole section (all transformed cells) in a group:

```ts
group(
    { fill: "white", "stroke-width": 1 },
    cells.map((cell) =>
        pathFromCubics(asCubic(polygon(simplify(cell, 0.01, true)), opts)
    )
),
```